### PR TITLE
Defensive editor

### PR DESCRIPTION
**Pull request by jpaddison3**
Adds several sections of defensive null checking.

More controversially, adds an error if you specifically request a version of a post you don't have access to. I'm not sure where @raemon and I wound up on this, but I think I had addressed his concerns.

The general case against: In general, when you request a document which a) you have access to and b) you don't have access to a field of, you get only the information you have access to. In the case where you do have access to the post but not the revision, therefore, you should get the post info, but not the contents of the version that you don't have access to.

Rebuttal in favor: Requesting a docId and a version is only done to specifically get the contents of that version. In that frame, it's more like the entire document is the thing you don't have access to. This also localizes the error close to the cause.

Specific objection from @Raemon: If you request all versions of a document (which will happen in the future) and a resolver throws an error, this probably leads to you getting *no* results, even for ones that you do have access to.

Rebuttal in favor: The code that requests all versions should either: a) get a list of versions it does have access to and request those. Specifically it should not be able to even find out the version number of a version it doesn't have access to. Or b) bypass the specific-version resolver in favor an approach that returns a list of revisions, silently excluding the ones for which the user does not have access.

### CODE REVIEW

**Comment by jpaddison3** [comment]
@Discordius @Raemon my guess is that this waiting for Ray to recover from his cold.

File: packages/lesswrong/components/editor/EditorFormComponent.jsx [hunk 1]
```diff
@@ -446,7 +449,10 @@ class EditorFormComponent extends Component {
 
   handleUpdateVersion = async (document) => {
     if (!document) return
-    const editorType = document.contents?.originalContents?.type
+    if (!document.contents?.originalContents) {
```
Comment (Raemon) [inline packages/lesswrong/components/editor/EditorFormComponent.jsx]: This error appears in console but doesn't isn't very visible to the user. (The user experiences "continuing to edit the latest version of the document". If they switch versions, nothing happens, but they can keep editing as if it's the latest one)

Comment (Raemon) [reply packages/lesswrong/components/editor/EditorFormComponent.jsx]: (would prefer to figure out how to propagate the error in a way that makes it more clear to the user something is broken)

**Comment by jpaddison3** [comment]
FYI this is about to get heavily back-burnered as I am about to be hit by Thanksgiving and then a team retreat and will have limited time until 2.5 weeks from now.

**Pull request closed without merging**